File: lib/features/dashboard/domain/entities/large_file_info.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'large_file_info.freezed.dart';
part 'large_file_info.g.dart';

@freezed
class LargeFileInfo with _$LargeFileInfo {
  const factory LargeFileInfo({
    required int id,
    required String name,
    required int size,
    required String mimeType,
    required int dateModified,
    required String uri,
    required String mediaType, // IMAGE, VIDEO, AUDIO, DOCUMENT, OTHER
  }) = _LargeFileInfo;

  factory LargeFileInfo.fromJson(Map<String, dynamic> json) =>
      _$LargeFileInfoFromJson(json);
}

File: android/app/src/main/kotlin/com/movixalabs/cleanmaster/models/LargeFileInfo.kt
package com.movixalabs.cleanmaster.models

data class LargeFileInfo(
    val id: Long,
    val name: String,
    val size: Long,
    val mimeType: String,
    val dateModified: Long,
    val uri: String,
    val path: String,
    val mediaType: MediaType
) {
    fun toMap(): Map<String, Any> = mapOf(
        "id" to id,
        "name" to name,
        "size" to size,
        "mimeType" to mimeType,
        "dateModified" to dateModified,
        "uri" to uri,
        "path" to path,
        "mediaType" to mediaType.name
    )
}

enum class MediaType {
    IMAGE, VIDEO, AUDIO, DOCUMENT, OTHER
}
